cfme/tests/infrastructure/test_vm_migrate.py

```diff
@@ -74,7 +76,37 @@ def test_vm_migrate_should_create_notifications_when_migrations_fail():
         initialEstimate: 1/4h
         startsin: 5.10
         tags: service
+        caseposneg: negative
+        testSteps:
+            1. add provider to appliance
+            2. create vm
+            3. migrate vm, use same host and datastore is None
+        expectedResults:
+            1.
+            2.
+            3. check error on UI and in the log file.
     Bugzilla:
         1478462
     """
-    pass
+    view = navigate_to(create_vm, 'Details')
+    vm_host = view.entities.summary('Relationships').get_text_of('Host')
+    hosts = [vds.name for vds in provider.hosts.all() if vds.name in vm_host]
+    migrate_to = hosts[0]
+    err_msg = (f'The VM \'{create_vm.name}\' can not be migrated to the same host it is '
+               'already running on.]')
+    with LogValidator('/var/www/miq/vmdb/log/automation.log',
+                      matched_patterns=[err_msg],
+                      hostname=appliance.hostname).waiting(timeout=900):
+        create_vm.migrate_vm("email@xyz.com", "first", "last", host=migrate_to, datastore="<None>")
+        request_description = create_vm.name
```

Comment:
We can use **fauxfactory** to generate emails and strings.

Reply:
Why cant we use **fauxfactory** for these fields: ```"first", "last",```?

Reply:
Done